metal/datasource_metal_facility.go

```diff
@@ -17,6 +17,13 @@ func dataSourceMetalFacility() *schema.Resource {
 				Description: "The code of the Facility to match",
 				Required:    true,
 			},
+			"features_required": {
+				Type:        schema.TypeSet,
+				Description: "Features which the facility needs to have.",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Optional:    true,
+				MinItems:    1,
```

Comment:
Since this field is optional, I think `MinItems: 0` is more fitting (https://github.com/equinix/terraform-provider-metal/pull/179/files#diff-df84d3f7d9e865eacdfcba301894d3db8d7c40830f29e13cd7cf02b585eb8a6bR65 looks like it will work as-is). I can imagine a configuration that takes user variables, perhaps in a map, where each server or facility has different feature requirements. Some of those servers or facilities may not need any features while others do.  

It would be more convenient to do:

```
data "metal_facility" "test" {
  count = len(var.configs)

  code = var.configs[count.index].facility.code
  features_required = var.configs[count.index].facility.features
}
```

than:

```
data "metal_facility" "test" {
  count = len(var.configs)

  code = var.configs[count.index].facility.code
  features_required = len(var.configs[count.index].facility.features) > 0 ? var.configs[count.index].facility.features : null
}
```

Although the user could just as easily put `null` in their `var.configs[].facility.features`.

Reply:
We can revisit this later.